src/components/EntryForm.jsx

```diff
@@ -0,0 +1,288 @@
+              className="capitalize shrink-0"
+              size="sm"
+              endContent={
+                <ChevronDownIcon className={"w-4 h-4 text-inherit"} />
+              }
+            >
+              {selectedProject || "Project"}
+            </Button>
+          </DropdownTrigger>
+          <DropdownMenu
+            aria-label="Single selection example"
+            variant="flat"
+            disallowEmptySelection
+            selectionMode="single"
+            selectedKeys={selectedKeys}
+            onSelectionChange={setSelectedKeys}
+            disabledKeys={projects === null ? ["__no_project__"] : []}
+            onAction={() => setProjectError(false)}
+          >
+            {projects !== null ? (
+              projects.map((project) => (
+                <DropdownItem key={project.name} className="capitalize">
+                  {formatProjectName(project.name)}
+                </DropdownItem>
+              ))
+            ) : (
+              <DropdownItem key="__no_project__" className="text-center">
+                <Spinner size="sm" color="primary" />
+              </DropdownItem>
+            )}
+          </DropdownMenu>
+        </Dropdown>
+      </div>
+
+      <div className="flex gap-1 items-center">
+        <TimeInput
+          variant="bordered"
+          radius="sm"
+          className="w-min"
+          aria-label="Start Time"
+          size="sm"
+          value={startTime}
+          onChange={(time) => {
+            startDateTime.hour = time.hour;
+            startDateTime.minute = time.minute;
+            setStartDateTime(startDateTime.copy());
+          }}
+        />
+        <DashIcon className="w-2 h-2 text-default-300" />
+        <Badge
+          content={
+            totalTime.duration >= 24 * 60 * 60 * 1000
+              ? Math.floor(totalTime.duration / (1000 * 60 * 60 * 24))
+              : timeDifference < 0
+              ? 1
+              : null
+          }
+          variant="faded"
+          color="default"
+          className="w-min"
+          isInvisible={
+            !(totalTime.duration >= 24 * 60 * 60 * 1000
+              ? true
+              : timeDifference < 0
+              ? true
+              : false)
+          }
+        >
+          <TimeInput
+            variant="bordered"
+            radius="sm"
+            aria-label="End Time"
+            className="w-min"
+            value={endTime}
+            size="sm"
+            onChange={(time) => {
+              endDateTime.hour = time.hour;
+              endDateTime.minute = time.minute;
+              const difference = endDateTime.compare(startDateTime);
+              if (difference < 0) {
+                endDateTime.day = startDateTime.day + 1;
+              } else if (
+                difference >= 24 * 60 * 60 * 1000 &&
+                difference < 48 * 60 * 60 * 1000
+              ) {
+                endDateTime.day = startDateTime.day;
+              }
```

Comment:
(suggestion; non-blocking): It's common practice to create boolean expressions with complex conditions to help with readability. For example:

```JavaScript
// Define boolean variables for different conditions
const isBeforeStart = timeDifference < 0;
const isBetween24And48Hours = timeDifference >= 24 * 60 * 60 * 1000 && timeDifference < 48 * 60 * 60 * 1000;

// Adjust endDateTime's day based on the time difference
if (isBeforeStart) {
  // If endDateTime is before startDateTime, increment endDateTime's day
  endDateTime.day = startDateTime.day + 1;
} else if (isBetween24And48Hours) {
  // If the time difference is between 24 and 48 hours, keep endDateTime's day same as startDateTime's day
  endDateTime.day = startDateTime.day;
}
```